### DISCUSSION TITLE
CableConstraint

### DISCUSSION THREAD

**Discussion by james12parker** [SofaDefrost/SoftRobots#239] [Help - Programming with SoftRobots] [answered]
Hello!

I have created an object and defined the object with a Tetrahedron. I am hoping to simulate this object and observe the deformation.

While exploring several examples, it appears that most people use a CableConstraint to control the soft robot.

However, when I attempted to implement the CableConstraint, I encountered an error stating:
```
Python exception:   
  ValueError: Object type CableConstraint<> was not created  
The object 'CableConstraint' is not in the factory.  
But the following exits:  
                      : ParabolicConstraint (76% match)  
                      : FixedConstraint (73% match)  
                      : AttachConstraint (70% match)  
                      : PartialFixedConstraint (70% match)  
                      : SlidingConstraint (68% match)  
                      : StopperConstraint (68% match)
```

My snippet code is:
```
def addActuator(parentNode):
        actuatorNode=parentNode.addChild("actuator")
        actuatorNode.addObject("MechanicalObject", position="3 -12 -20 3 -12 -10 10 -12 10 10 -12 30 10 -12 45")
        actuatorNode.addObject("CableConstraint", indices="0 1 2 3 4", pullPoint="3 -12 -100", value="40", valueType="displacement")
        actuatorNode.addObject("BarycentricMapping", mapForces="false", mapMasses="false")
```

Am I using the CableConstraint incorrectly?

Thank you.

**Comment by alxbilger** [comment, marked as answer]
My guess is that you did not load the SoftRobots plugin. See the [documentation page](https://www.sofa-framework.org/community/doc/plugins/what-is-a-plugin/#plugin-loading) if you need to know how to do it.
